crates/wasmtime/src/runtime/ref/no_gc_ref.rs

```diff
@@ -0,0 +1,60 @@
+//! The dummy `ExternRef` type used when the `gc` cargo feature is disabled.
+//!
+//! Providing a dummy type means that downstream users need to do fewer
+//! `#[cfg(...)]`s versus if this type or its methods simply didn't exist. The
+//! only methods that are left missing are constructors.
+
+#![allow(missing_docs)]
+
+use crate::runtime::Uninhabited;
+use crate::AsContextMut;
+use std::any::Any;
+use std::ffi::c_void;
+use wasmtime_runtime::VMExternRef;
+
+/// Represents an opaque reference to any data within WebAssembly.
+///
+/// Due to compilation configuration, this is an uninhabited type: enable the
+/// `gc` cargo feature to properly use this type.
+#[derive(Clone, Debug)]
+pub struct ExternRef {
+    pub(crate) _inner: Uninhabited,
+}
+
+impl ExternRef {
+    pub(crate) fn from_vm_extern_ref(_inner: VMExternRef) -> Self {
+        unreachable!()
```

Comment:
Could this perhaps to `_inner.assert_unreachable()` or something like that? (where `VMExternRef` only has that method when the `gc` feature is disabled)